torch/_inductor/ir.py

```diff
@@ -2347,8 +2347,13 @@ def copy_input(x):
 
     @classmethod
     def process_kernel(cls, kernel, *args, **kwargs):
-        binded_args = signature(kernel).bind(*args, **kwargs).arguments
-        args_flat, args_spec = pytree.tree_flatten(binded_args)
+        # Define a wrapper so that the type signature always contains args and
+        # kwargs, which get accessed in unflatten_args.
+        def generic_kernel(*args, **kwargs):
+            return kernel(*args, **kwargs)
+
+        bound_args = signature(generic_kernel).bind(*args, **kwargs).arguments
```

Comment:
Note: should also call `generic_kernel` where `kernel` is used below, just to be safe.

```
example_output = kernel(*new_args, **new_kwargs)
```